api/app/src/routes/medical/organization.ts

```diff
@@ -1,41 +1,55 @@
 import { Request, Response } from "express";
 import Router from "express-promise-router";
+
+import { organizationSchema } from "./models/organization";
 import { asyncHandler, getCxIdOrFail } from "../util";
 const router = Router();
 import status from "http-status";
 import { updateOrganization } from "../../command/medical/organization/update-organization";
 import { createOrganization } from "../../command/medical/organization/create-organization";
 import { getOrganization } from "../../command/medical/organization/get-organization";
-import { Organization } from "../../models/medical/organization";
+import { Organization as LocalOrg } from "../../models/medical/organization";
+import { Organization } from "./models/organization";
+import { createOrUpdateCWOrg } from "../../external/commonwell/organization";
 
 /** ---------------------------------------------------------------------------
  * POST /organization
  *
  * Updates or creates the organization if it doesn't exist already.
  *
- * @return  {Organization}  The organization.
+ * @return  {LocalOrg}  The organization.
  */
 router.post(
   "/",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
 
-    // TODO: parse this into model
-    const orgData = req.body;
+    const reqOrgData = organizationSchema.parse(req.body);
+
+    const localOrgPayload: Organization = {
+      name: reqOrgData.name,
+      type: reqOrgData.type,
+      location: reqOrgData.location,
+    };
+
+    await createOrUpdateCWOrg({
+      name: reqOrgData.name,
+      orgId: reqOrgData.id,
+      localOrgPayload,
+    });
 
     // update if this is an existing org
-    let org: Organization;
-    if (orgData.id) {
-      const data = { ...orgData };
+    let localOrg: LocalOrg;
+
+    if (reqOrgData.id) {
+      const data = { ...reqOrgData };
       delete data.id;
-      org = await updateOrganization({ id: orgData.id, cxId, data });
+      localOrg = await updateOrganization({ id: reqOrgData.id, cxId, data: localOrgPayload });
```

Comment:
We were passing `data` here bc we didn't want to send the `id` to the `data` field of the org on the DB. This is likely just an undue update, right?

Reply:
I assume so yea. What do you mean by undue update?

Reply:
Oh, I was assuming it was one of those "copy/paste" like situations because the "else" part of the "if" also had a similar update. 😄